### PR TITLE
[PLAY-203] Full-height Variant for Dialog

### PR DESCRIPTION
**Pull request by augustomallmann**
____

#### Screens

Screen Shot 2022-09-27 at 2 15 41 PM
Screen Shot 2022-09-27 at 2 16 01 PM

#### Breaking Changes

No breaking changes, introducing two new props : `fullHeight` and `placement`.

Note that `border-radius` has changed to `none` for all dialogs.

#### Runway Ticket URL

[[PLAY-203] 
](https://nitro.powerhrg.com/runway/teams/432/backlog/sprints/active#open-backlog-item-140805)

#### How to test this

- Check for the acceptance criteria under on Runway  

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [x] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_dialog/child_kits/_dialog_footer.tsx [hunk 1]
```diff
@@ -41,9 +38,9 @@ const DialogFooter = (props: DialogFooterProps) => {
 
   return (
     <>
-      <If condition={separator}>
-        <SectionSeparator />
-      </If>
+      {separator &&
+        <SectionSeparator aria={{dd: 'ff'}} className="dd" data={{dd: 'ff'}} id="d" text="ss"/>
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/child_kits/_dialog_footer.tsx:42]: What are these attributes?

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_full_height.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+The full height dialog can be size Medium, Large and XLarge.
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_full_height.md:1]: Did you mean to say small, medium, and large?

File: playbook/app/pb_kits/playbook/pb_dialog/child_kits/_dialog_footer.tsx (hunk 1 above)
Comment (augustomallmann) [reply playbook/app/pb_kits/playbook/pb_dialog/child_kits/_dialog_footer.tsx:42]: Good catch, they were not supposed to be there, were used for a test only. thanks!

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_full_height.md (hunk 2 above)
Comment (augustomallmann) [reply playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_full_height.md:1]: Yes, the text is incorrect. thanks!

**Review by thestephenmarshall — APPROVED** [review]

File: playbook/package.json [hunk 3]
```diff
@@ -87,6 +87,7 @@
     "@types/lodash": "4.14.186",
     "@types/react-modal": "^3.13.1",
     "@types/react": "17.0.50",
+    "@types/react-modal": "^3.13.1",
```
Comment (jasperfurniss) [inline playbook/package.json:90]: This looks like a repeat of line 88, two lines above. @augustomallmann

File: playbook/app/pb_kits/playbook/pb_dialog/_dialog.scss [hunk 4]
```diff
@@ -135,5 +137,34 @@
 			animation-duration: $animation-duration;
 			opacity: $opacity_hidden;
 		}
+		&[class*="full_height"] {
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_dialog/_dialog.scss:140]: Any idea why it's indenting so much on the scss here?

File: playbook/app/pb_kits/playbook/pb_dialog/_dialog.scss [hunk 5]
```diff
@@ -135,5 +137,34 @@
...
+					width: $medium;
+				}
+				&[class*="_md"] {
+					width: $large;
+				}
+				&[class*="_lg"] {
+					width: $xlarge;
+				}
+			}
+		}
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_dialog/_dialog.scss:168]: indentation seems weird here.

**Pull request merged**